AttackItem.h

```diff
@@ -3,7 +3,8 @@
 
 #include "Item.h"
 #include "Character.h"
-#include "Utility.h"
+
+struct Character;
```

Comment:
Don't include Character.h if you're going to forward declare it..
Include it in the cpp file, since that's where you use it.